documents/error.go

```diff
@@ -8,6 +8,12 @@ import (
 
 const (
 
+	// ErrDocumentConfigTenantID must be used for errors related to tenantID operations
```

Comment:
Aiming to have this PR merged asap, is it possible to take this error part out into another one? As it will bring up more questions as shown below. Or @vedhavyas we are ok keeping this err definition, and change it later?

Reply:
- We discussed error types
- Agreed on an implementation
- Now we have the implementation
- I'm using it, and now its about which place to use it. My preference is not to make things so subjective and flexible so that everyone chooses whether to add an error type or not based on personal judgement. By default we go with error types and we decide later when we see a particular error type not being used much and remove, its an easy refactoring.

Reply:
@vedhavyas ^^ I am ok either way.